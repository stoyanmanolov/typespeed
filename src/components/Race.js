import React from "react";
import { Input, Progress } from "semantic-ui-react";

const paragraphs = [
  "The word “Retro” comes from Latin word retro, meaning backward or past times. Retro style fonts are outdated or aged style fonts that imply a vintage of at least 15 or 20 years. Texts in retro style can take you or your audience to the good old memories."
];

class Race extends React.Component {
  state = { text: "", userInput: "" };

  componentDidMount = () => {
    let randomParagraph =
      paragraphs[Math.floor(Math.random() * paragraphs.length)];
    this.setState({ text: randomParagraph });
  };

  handleChange = event => {
    this.setState({ userInput: event.target.value });
  };

  render() {
    return (
      <div className="container">
        <h2>Better go fast...</h2>
        <Progress id="progress-bar" percent="11" />
        <p className="race-text">{this.state.text}</p>
        <div className="typearea">
          <label>Type here</label>
          <Input>
            <input
              value={this.state.userInput}
              id="input-field"
              type="text"
              onChange={this.handleChange}
            />
          </Input>
        </div>
      </div>
    );
  }
}

export default Race;
