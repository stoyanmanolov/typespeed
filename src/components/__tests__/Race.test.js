import React from "react";
import { shallow } from "enzyme";
import Race from "../Race";

jest.useFakeTimers();

describe("The Race component", () => {
  let wrapper = shallow(<Race />);

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("when the component mounts", () => {
    it("the paragraph text and its length in state are not empty", () => {
      expect(wrapper.state().text.length).toBeGreaterThan(0);
      expect(wrapper.state().initialTextLength).toBeGreaterThan(0);
    });

    it("the seconds timer runs correctly", () => {
      let seconds = 5;
      let milliseconds = seconds * 1000;
      jest.advanceTimersByTime(milliseconds);
      expect(wrapper.state().seconds).toEqual(seconds);
    });
  });
  describe("when the component unmounts", () => {
    it("stops the timer", () => {
      wrapper.unmount();
      expect(clearInterval).toHaveBeenCalled();
    });
  });
});
