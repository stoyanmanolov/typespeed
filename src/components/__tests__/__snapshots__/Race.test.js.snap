// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Race component renders correctly 1`] = `ShallowWrapper {}`;
